src/common/languageinfo.cpp

```diff
@@ -56,6 +58,47 @@ static std::unordered_map<wxString, wxString> gs_scmap_alias2name;
         return wxString();
 }
 
+// Hash map to look up likely subtags
+static std::unordered_map<wxString, wxString> gs_likely_subtags_map;
+
+/* static */ wxString wxUILocaleImpl::GetLikelySubtags(const wxString& fromTag)
+{
+    const auto tagIter = gs_likely_subtags_map.find(fromTag);
+    if (tagIter != gs_likely_subtags_map.end())
+        return tagIter->second;
+    else
+        return wxString();
+}
+
+// Hash map to look up distances for matching tag pairs
+static std::unordered_map<wxString, int> gs_matching_tags_map;
+
+/* static */ int wxUILocaleImpl::GetMatchDistance(const wxString& desiredTag, const wxString& supportedTag)
+{
+    const auto tagIter = gs_matching_tags_map.find(desiredTag+wxString(":")+supportedTag);
+    if (tagIter != gs_matching_tags_map.end())
+        return tagIter->second;
+    else
+        return -1;
+}
+
+// Hash map to look up distances for matching tag pairs
+static std::unordered_map<wxString, std::unordered_set<wxString>> gs_region_groups_map;
+
+/* static */ bool wxUILocaleImpl::SameRegionGroup(const wxString& language, const wxString& desiredRegion, const wxString& supportedRegion)
+{
+    const auto languageIter = gs_region_groups_map.find(language);
+    if (languageIter != gs_region_groups_map.end())
+    {
+        const auto groupSet = languageIter->second;
+        bool containsDesired = groupSet.find(desiredRegion) != groupSet.end();
+        bool containsSupported = groupSet.find(supportedRegion) != groupSet.end();
+        return (containsDesired && containsSupported) || (!containsDesired && !containsSupported);
```

Comment:
This condition is difficult to read/understand IMO. Maybe writing it like this:

```suggestion
        if (containsDesired && containsSupported)
            return true;
            
        if (!containsDesired && !containsSupported)
            return true;
```

could help a bit, especially if comments explaining what do the 2 cases correspond to are added.

Reply:
I add a comment in front of the original code. IMHO splitting the condition doesn't make it clearer and better to understand. Desired and supported region belong to the same region group, if they are either both **contained** or both **not contained** in the checked region group.